app.ts

```diff
@@ -196,6 +202,37 @@ setInterval(async () => {
       let memberData = memberDoc.data() as MemberData
       if (memberData[date].timeStamp === undefined) {
         let reporterName = memberData[date].reporter
+        const optKeyboard: telegramBot.SendMessageOptions = {
+          parse_mode: "Markdown",
+          reply_markup: {
+            keyboard: [[{ text: "Source Did not respond yet" }], [{ text: "Did not asked the report yet" }]]
+          }
+        }
+        const optRemove: telegramBot.SendMessageOptions = {
+          reply_markup: {
+            remove_keyboard: true
+          }
+        }
+        await bot.sendMessage(
+          memberNameChatIdMap[reporterName],
+          `Just in case your source didn't show up, let us know.`,
+          optKeyboard
+        )
+        var chatId = parseInt(memberNameChatIdMap[reporterName])
+        responseCallbacks[chatId] = async response => {
+          var status = (response.text == "Source Did not respond yet") ? true : false
```

Comment:
As I understand this, User gets a message asking whether source showed up yet or they have to contacted the source yet and after that we send a message saying "just in case source didn't show up, let us know" and we leave the bot hanging and expecting a reply until it sends a message to them again?

I was thinking that this could just be a simple question asked to the user and they could press on "yes" or "no" buttons instead of taking in a sentence input and match it verbatim with "Source Did not respond yet"

Reply:
cool, I'll change it .